docs/codeql/codeql-cli/about-codeql-packs.rst

```diff
@@ -63,41 +59,46 @@ The following properties are supported in ``qlpack.yml`` files.
      - Required
      - Purpose
    * - ``name``
-     - ``codeql/queries``
+     - ``octo-org/security-queries``
      - All packs
-     - The scope, where the CodeQL pack is published, and the name of the pack defined using alphanumeric characters, hyphens, and periods. It must be unique as CodeQL cannot differentiate between CodeQL packs with identical names. Use the pack name to specify queries to run using ``database analyze`` and to define    dependencies between QL packs (see examples below).
+     - The scope, where the CodeQL pack is published, and the name of the pack defined using alphanumeric characters and hyphens. It must be unique as CodeQL cannot differentiate between CodeQL packs with identical names. Name components cannot start or end with a hyphen. Additionally, a period is not allowed in pack names at all. Use the pack name to specify queries to run using ``database analyze`` and to define dependencies between QL packs (see examples below).
    * - ``version``
      - ``0.0.0``
      - All packs
      - A version number for this CodeQL pack. This must be a valid semantic version that meets the `SemVer v2.0.0 specification <https://semver.org/spec/v2.0.0.html>`__.
    * - ``dependencies``
-     - ``github/codeql-javascript:1.2.3``
+     - ``codeql/javascript-all: 1.2.3``
      - Optional
-     - The names of any CodeQL packs that this pack depends on, as a sequence. This gives the pack access to any libraries, database schema, and query suites defined in the dependency.
-   * - ``default-suite-file``
-     - ``default-queries.qls``
+     - The names of any CodeQL packs that this pack depends on, as a sequence. This gives the pack access to any libraries, database schema, and query suites defined in the dependency. Optionally a version for the dependency is specified. 
```

Comment:
This is not quite right. It is not a _sequence_, but a _mapping_ (sequence implies an order, whereas a mapping is a relation from keys to values).

Also, the version range is required. And the value can be any semver compliant version range. A good description is here: https://docs.npmjs.com/cli/v6/using-npm/semver#ranges

It's not the canonical description, but not sure if there is one. You can include this link in the docs for a good primer on semver ranges.

Reply:
I have changed this to state that the version range is required but this is not what the help for the `pack add` command says for CodeQL CLI 2.5.8:

```
This command modifies the qlpack.yml file of the current package to add one or more dependencies on the specified QL library pack.

To specify a version range instead, please edit the qlpack.yml file directly.


      <scope/name[@version]>...
                             [Mandatory] The scope, name, and optional version of the pack to add to the dependency list.
                             If the version is missing, then the latest version of this pack is retrieved.
```

Are we saying that if you add a dependency to a CodeQL pack, you don't need to include a version because we add one for you, but that if you edit the file then you must add a version range?

Reply:
Thank you for catching that. The cli docs are outdated.

The behaviour is that you can specify a version range when running `codeql pack add` and that range will be placed directly into the qlpack.yml. The version installed will be the latest version that satisfies the range. This is also the version placed in the lock file.

Note that all semver versions are valid semver ranges.

I will create a PR to fix the cli docs.

Reply:
https://github.com/github/semmle-code/pull/39843